packages/web-components/fast-components/src/design-system-consumer/design-system-consumer.ts

```diff
@@ -1,91 +1,38 @@
-import { observable } from "@microsoft/fast-element";
-import { DesignSystemResolverEntry } from "../styles/recipes";
+import { Behavior } from "@microsoft/fast-element";
 import { composedParent } from "../utilities";
 import { DesignSystemProvider } from "../design-system-provider";
 
-export interface ConsumerArgs {
-    recipes: DesignSystemResolverEntry[];
-}
-
 export interface DesignSystemConsumer {
-    recipes: DesignSystemResolverEntry[];
     provider: DesignSystemProvider | null;
-    isDesignSystemConsumer: boolean;
 }
 
 /**
- * Type-safe checking for if an HTMLElement is a DesignSystemProvider.
- * @param el The element to test
+ * Behavior to connect an element to the nearest design-system provider
  */
-export function isDesignSystemProvider(
-    el: HTMLElement | DesignSystemConsumer
-): el is DesignSystemConsumer {
-    return (el as any).isDesignSystemConsumer;
-}
-
-export function designSystemConsumer<T extends { new (...args: any[]) }>(
-    constructor: T
-): T;
-export function designSystemConsumer<T extends { new (...args: any[]) }>(
-    args: ConsumerArgs
-): <_T>(constructor: _T) => _T;
-export function designSystemConsumer<T extends { new (...args: any[]) }>(
-    argsOrConstructor: any
-): any {
-    function decorator(constructor: T, options: ConsumerArgs): T {
-        class Consumer extends constructor implements DesignSystemConsumer {
-            public readonly isDesignSystemConsumer = true;
-
-            public readonly recipes = options.recipes;
-
-            @observable
-            public provider: DesignSystemProvider | null = null;
-
-            /**
-             * Find the parent DesignSystem provider.
-             */
-            public findProvider(): DesignSystemProvider | null {
-                let parent = composedParent(this as any);
-
-                while (parent !== null) {
-                    if ((parent as any).isDesignSystemProvider) {
-                        return parent as any;
-                    } else if (isDesignSystemProvider(parent)) {
-                        return parent.provider;
-                    } else {
-                        parent = composedParent(parent);
-                    }
-                }
-
-                return null;
-            }
-
-            public connectedCallback(): void {
-                super.connectedCallback();
-
-                const provider = this.findProvider();
+export class DesignSystemConsumerBehavior<T extends DesignSystemConsumer & HTMLElement>
+    implements Behavior {
+    bind(source: T) {
+        source.provider = findProvider(source);
+    }
 
-                if (!!provider) {
-                    this.provider = provider;
-                    this.provider.subscribe(this);
-                }
-            }
+    /* eslint-disable-next-line */
```

Comment:
@EisenbergEffect It seems like the `bind` method is required but may not always be needed - perhaps this could be optional?

Reply:
It's tricky to make it optional because of the perf hit in checking for its existence. I'm planning some experiments around instantiate and bind related to perf. As part of that work, I could try out some ideas to remove unbind and maybe allow behaviors to add themselves to an unbind list via their bind implementation. I'm not sure. It's a big change to make. So for now, I think we need to stick with an empty unbind and then circle back around to see if we can improve that later.